### PR TITLE
Fix bug in benchmark binary file spiller where pagebuilder is not reset between accumulating pages

### PR DESCRIPTION
**Pull request by mbeitchman**
The pagebuilder should be reset so it
doesn't accumulate pages during each loop.

### CODE REVIEW

**Comment by mbeitchman** [comment]
addresses https://github.com/prestodb/presto/issues/8773

**Comment by mbeitchman** [comment]
I am on the Amazon CLA

**Comment by mbeitchman** [comment]
I am on corporate cla for Amazon.

**Comment by mbeitchman** [comment]
is this a test issue?

File: presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java [hunk 1]
```diff
@@ -142,6 +142,7 @@ public void tearDown()
                     DOUBLE.writeDouble(pageBuilder.getBlockBuilder(4), lineItem.getExtendedPrice());
                 }
                 pages.add(pageBuilder.build());
+                pageBuilder.reset();
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: If your intention indeed is to fix this single benchmark, please make it explicit in commit message and PR title

Comment (mbeitchman) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: Can you suggest a better title? I am a new contributor to Presto so will help to get your suggestion.

Comment (mbeitchman) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: Updated the title. Please let me know if that is better.

Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: 👍 , you could also update the commit message (but mind the guidelines https://chris.beams.io/posts/git-commit/). e.g

```
Fix PageBuilder use in benchmark binary file spiller

`PageBuilder` needs to be `reset()` between building pages.
```

**Comment by mbeitchman** [comment]
Thanks! Will update the commit message now.

File: presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java (hunk 1 above)
Comment (mbeitchman) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: Nice! thanks

**Comment by mbeitchman** [comment]
Commit message updated. Thanks for the help.

**Comment by findepi** [comment]
@mbeitchman please be sure to fill CLA. Legally, it should not be required for one-liner bug fix, but please fill it to be on the safe side.

**Comment by mbeitchman** [comment]
@findepi I am covered by Amazon CLA.

**Comment by mbeitchman** [comment]
Thanks!

**Pull request merged**